### ISSUE TITLE
Request constructor from JS should normalize URLs

### ISSUE THREAD

**Issue by yusukebe** [oven-sh/bun#3831]
### What version of Bun is running?

0.7.0

### What platform is your computer?

Darwin 22.5.0 arm64 arm

### What steps can reproduce the bug?

```ts
console.log(new Request('http://localhost/static/../favicon.ico').url)
```

will show:

```
http://localhost/static/../favicon.ico
```

### What is the expected behavior?

Should be `http://localhost/favicon.ico`.

Other runtimes - workerd, Deno, Node.js -  's Request return `http://localhost/favicon.ico`. If you allow `..`, there will be the possibility to cause such as vulnerabilities against directory traversal attack.

**Comment by Jarred-Sumner** [comment]
Yeah we aren't running the `Request` object through the URL parser currently. Need to fix that

**Comment by yusukebe** [comment]
Okay, thanks!

**Comment by moaoa** [comment]
I think we need to normalize the URL here  
https://github.com/oven-sh/bun/blob/413fd281208f24a532c47249dec1bfc3aef2bf37/src/http/url_path.zig#L34-L48

**Comment by Jarred-Sumner** [comment]
Fixed in #3902
